File: app/src/main/java/upvictoria/pm_ene_abr_2024/iti_271164/pg1u1_eq02/MainActivity.java
package upvictoria.pm_ene_abr_2024.iti_271164.pg1u1_eq02;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private MainSurfaceView mainSurfaceView;
    private Button buttonClass;
    private Button buttonLine;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mainSurfaceView = findViewById(R.id.surfaceview);
        buttonClass = findViewById(R.id.bt1);
        buttonLine = findViewById(R.id.bt2);

        buttonClass.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Set the flag to indicate drawing a figure
                mainSurfaceView.setDrawingFigure(true);
                mainSurfaceView.setDrawingLine(false);
            }
        });

        buttonLine.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Set the flag to indicate drawing a line
                mainSurfaceView.setDrawingFigure(false);
                mainSurfaceView.setDrawingLine(true);
            }
        });
    }
}

File: app/src/main/java/upvictortia/pm_ene_abr_2024/iti_271164/pi1u1/z_u1_mota_hinojosa/MainActivity.kt
package upvictortia.pm_ene_abr_2024.iti_271164.pi1u1.z_u1_mota_hinojosa

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    private lateinit var continueBtn: Button
    private lateinit var textView1: TextView
    private lateinit var textView2: TextView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        continueBtn = findViewById(R.id.btn_continue)
        textView1 = findViewById(R.id.text_view1)
        textView2 = findViewById(R.id.text_view2)
        continueBtn.setOnClickListener { view ->
            val toast = Toast.makeText(this,"",Toast.LENGTH_SHORT)
            try {
                if (textView1.text.isNotEmpty() || textView2.text.isNotEmpty()) {
                    val count1 = Integer.parseInt(textView1.text.toString())
                    val count2 = Integer.parseInt(textView2.text.toString())
                    if (count1 > 0 && count2 > 0) {
                        val intent = Intent(this,InputActivity::class.java).apply {
                            putExtra("COUNT1",count1)
                            putExtra("COUNT2",count2)
                        }
                        startActivity(intent)
                    }
                    else {
                        throw NumberFormatException("Número Inválido")
                    }
                } else {
                    throw IllegalArgumentException("Ningún campo debe de estar vacío")
                }
            } catch (e : IllegalArgumentException) {
                showToast(e.message.toString(),toast)
            } catch (e : NumberFormatException) {
                showToast(e.message.toString(),toast)
            }
        }
    }

    private fun showToast(msg : String, toast : Toast) {
        toast.setText(msg)
        toast.show()
    }
}